### PR TITLE
Ignore parallel option in fast C reader

### PR DESCRIPTION
**Pull request by astrofrog**
As discussed in https://github.com/astropy/astropy/issues/8858, this ignores for now the parallel option in the fast C reader in io.ascii, since serial reading has better performance. The warning is deliberately not a deprecation warning as this may be fixed in future.

This does mean that we now have some code that isn't tested and I think it's important that we simply remove this code if the bug isn't fixed in a timely way, so I'll open an issue to remind us to remove this code, and I'll milestone it for 4.2 initially though we could discuss what a good time frame is. I think it makes sense to be reasonably aggressive or we'll forget about it. In any case we should keep this present PR simple and we can backport it to LTS and 4.1.

Running the test suite before this PR on MacOS X with Python 3.8:

```
% pytest astropy/io/ascii                            
...
======= 547 passed, 27 skipped, 6 xfailed in 293.08s (0:04:53) =======
```

and with this PR: 

```
======= 547 passed, 27 skipped, 6 xfailed in 18.73s =======
```

Fixes https://github.com/astropy/astropy/issues/9864

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -953,6 +953,11 @@ astropy.extern
 astropy.io.ascii
 ^^^^^^^^^^^^^^^^
 
+- Partially fixed a performance issue when reading in parallel mode. Parallel
+  reading does not work correctly and has worse performance than serial
```
Comment (taldcroft) [inline CHANGES.rst:957]: When you say "does not work correctly" that could imply it is returning the wrong answer. I did not see any statements this the answer is wrong, just that it is slow. If so, I would reword this here and elsewhere saying that "Parallel reading currently has substantially worse performance than the default serial reading, so we now ignore ...".

File: astropy/io/ascii/cparser.pyx [hunk 2]
```diff
@@ -219,6 +219,17 @@ cdef class CParser:
...
 
+        # FIXME: for now the parallel mode does not work correctly and is worse
+        # than non-parallel mode so we disable parallel mode if set and emit a
+        # warning. We keep the parallel code below so that it can be fixed in
+        # future, but if it cannot be fixed we should remove the parallel code
+        # and deprecate the option itself. For now the warning is not a
+        # deprecation warning since we may still fix it in future.
+        if parallel:
+            warnings.warn('parallel reading does not currently work, '
+                          'so falling back to serial reading', AstropyWarning)
```
Comment (taldcroft) [inline astropy/io/ascii/cparser.pyx:230]: I think it might help anybody that hits this to have a reference to issue #8858 in the warning text.

**Comment by taldcroft** [comment]
About the milestone for removing the code, since the 4.2 feature freeze is 7 days away that seems entirely unrealistic. I'd just start off with the milestone of 4.3 for either fixing or removing the parallel code.

File: CHANGES.rst (hunk 1 above)
Comment (astrofrog) [reply CHANGES.rst]: Done

File: astropy/io/ascii/cparser.pyx (hunk 2 above)
Comment (astrofrog) [reply astropy/io/ascii/cparser.pyx]: Done

**Comment by astrofrog** [comment]
@taldcroft - I think your comments are addressed now

**Review by taldcroft — APPROVED** [review]
LGTM
